sinfonier-web/app/models/module/ModuleVersion.java

```
@@ -261,44 +261,49 @@ public void recheck() throws SinfonierException {
     save();
   }
 
-  public File exportAsJson(Module module) {
-    DBObject obj = module.toDBObject();
-    obj.putAll(this.toDBObject());
-
-    obj.removeField(FIELD_ID);
-    obj.removeField(FIELD_ICON);
-    obj.removeField(FIELD_STATUS);
-    obj.removeField(FIELD_AUTHOR_ID);
-    obj.removeField(FIELD_CREATED);
-    obj.removeField(FIELD_UPDATED);
-    obj.removeField(FIELD_RATINGS);
-    obj.removeField(FIELD_AVERAGE_RATE);
-    obj.removeField(FIELD_TOPOLOGIES_COUNT);
-    obj.removeField(FIELD_MY_TOOLS);
-    obj.removeField(FIELD_COMPLAINS);
-    obj.removeField(FIELD_CONTAINER);
-    obj.removeField(FIELD_CATEGORY);
-    obj.removeField(FIELD_VERSIONS);
-
-    JsonObject res = Mongo2gson.getAsJsonObject(obj);
-    File dir = FileUtils.getFile(PATH_TO_SAVE);
-
-    if (!dir.exists()) {
-      dir.mkdirs();
-    }
-
-    String fileName = module.getName() + ".json";
-    File file = FileUtils.getFile(dir, fileName);
-
-    try {
-      FileUtils.writeByteArrayToFile(file, res.toString().getBytes("UTF-8"));
-    } catch (IOException e) {
-      Logger.error(e.getMessage());
-    }
-
-    return file;
-  }
-
+	public File exportAsJson(Module module) {
+		JsonObject res = toJson(module);
+		File dir = FileUtils.getFile(PATH_TO_SAVE);
+
+		if (!dir.exists()) {
+			dir.mkdirs();
+		}
+
+		String fileName = module.getName() + ".json";
+		File file = FileUtils.getFile(dir, fileName);
+
+		try {
+			FileUtils.writeByteArrayToFile(file, res.toString().getBytes("UTF-8"));
+		} catch (IOException e) {
+			Logger.error(e.getMessage());

```

If an error happens this code only print a log. You should return an error.

---

ok